Implement decapitation victory condition

lL1l1 opened the pull request:
<!-- General useful tooling:
    - [ScreenToGif](https://www.screentogif.com/): Free, open source screen recorder that can export to MP4. If the changes are visual, these can help you tell us exactly what the changes imply!
-->
<!-- Feel free to remove unused parts of this template. -->

## Description of the proposed changes
Implement the decapitation victory condition, where all commanders in a team must die for the team to be defeated. Resolves #5347.

Still needs translation for other languages.

## Testing done on the proposed changes
<!-- List all relevant testing that you've done to confirm the changes work. -->
How it looks in the lobby UI:
![{16FA0A28-0A3F-4180-B25B-C1BAAA7EEF42}](https://github.com/user-attachments/assets/e77fcfc5-d0e0-4a72-8c47-1355430319d9)

What the log looks like when all brains are defeated:
```
info: { -- table: 20847D98 (48 bytes)} -- 1 ai is missing the ACU but is not defeated because of the alliance defeat requirement
info: { -- table: 272F1190 (48 bytes)}
info: { -- table: 1CEEE550 (72 bytes) -- both ais are now defeated, so the table fills
info:   {...} -- table: 205A6208 (2600 bytes),
info:   [3] = {...} -- table: 2065D730 (2600 bytes)
info: }
info: { -- table: 2722F938 (72 bytes) -- second call to the function to "take into account cascading effects"
info:   {...} -- table: 205A6208 (2600 bytes),
info:   [3] = {...} -- table: 2065D730 (2600 bytes)
info: }
debug: Matchstate - defeated: Thel-Awloshi (AI: Easy)
debug: Matchstate - defeated: Ronald (AI: Easy)
debug: Matchstate - won: Oum-Ullehesh (AI: Easy)
debug: Matchstate - won: Player
```

Defeat works when alliances are changed. Ran `SimLua SetAlliance(1, 3, 'enemy')` in the console to test. The game continues as expected afterwards (since I set myself to 1 team, my ally army 3 to another, and armies 2/4 to a third team).

## Additional Context
I did look at the mod by Sheikah but went with a different implementation that doesn't iterate unit checking over all brains multiple times.

## Checklist
- [x] Changes are annotated, including comments where useful
- [x] Changes are documented in the changelog for the next game version

lL1l1 committed:
Implement decapitation victory condition

lL1l1 committed:
Debug logging

lL1l1 committed:
Revert "Debug logging"

This reverts commit 7c2895606db6b1486a312ecb1e5e7ed2757d7c41.

lL1l1 committed:
Add comment

lL1l1 committed:
Create features.6667.md

lL1l1 committed:
Fix ArmyBrains global annotation

lL1l1 committed:
Support TransferToKiller share condition with decapitation

This is the only place I found the string `demoralization`

lL1l1 committed:
Update strings_db.lua

Hdt80bro commented:
My first thought was "uh, all commanders on a team needing to be destroyed for the team to be out is normal assassination" - but what this actually does is only make an army go out once all other teammates' commanders also get destroyed? If so, we should clarify that.

lL1l1 commented:
I was matching it with the simple language used in other victory conditions, where "Game Ends" means "Your army is put out of the game" and 2nd person isn't used, but you're right it's confusing that way.

lL1l1 committed:
Re-word description

lL1l1 committed:
Use loc tag not used in non-US loc

130-151 are used for other options such as prebuilt units, but only in the other language localization files. 152 and 153 are not used.

Hdt80bro commented on loc/US/strings_db.lua:
```diff
@@ -4533,6 +4533,8 @@ lobui_0126="Annihilation"
 lobui_0128="Sandbox"
 lobui_0129="Game never ends"
+lobui_0152="Decapitation"
+lobui_0131="Game ends when all allied commanders are destroyed"
```
Wrong tag

lL1l1 replied:
Fixed

lL1l1 committed:
Fix tag

lL1l1 committed:
Update features.6667.md
